gitlab-pipeline/stage/test.yml

```
@@ -623,20 +614,41 @@ test_full_integration_enterprise:
 
     # Output storage io stats
     - df -h . | tail -1 | awk '{system("hdparm -tT "$1);}'
+
     # Get and install the integration test requirements
     - apk add $(cat ${WORKSPACE}/integration/tests/requirements/apk-requirements.txt)
-    - pip install  -r ${WORKSPACE}/integration/tests/requirements/python-requirements.txt
-    - pip3 install -r ${WORKSPACE}/integration/tests/requirements/python-requirements.txt
+    - pip install -r ${WORKSPACE}/integration/tests/requirements/python-requirements.txt
 
-    # Load all docker images
-    - for repo in `integration/extra/release_tool.py -l docker`; do
+        # Load all docker images, and the client images depending on $BUILD_CLIENT
+    - |-
+        for repo in `integration/extra/release_tool.py -l docker`; do
+        # Only load the client in the case it is built for this pipe
+            if echo $repo | grep -q mender-client; then
+                if [ "${BUILD_CLIENT}" = "true" ]; then
+                    docker load -i stage-artifacts/${repo}.tar;
+                else
+                    continue
+                fi
+            fi
```

A matter of taste, I guess. But this syntax has two disadvantages IMO:
1) requires `;` at the end of commands (easy to miss)
2) it gets truncated in GitLab output, which makes quite harder the debug

I instead would use:
```suggestion
    - for repo in `integration/extra/release_tool.py -l docker`; do
    -    # Only load the client in the case it is built for this pipe
    -        if echo $repo | grep -q mender-client; then
    -            if [ "${BUILD_CLIENT}" = "true" ]; then
    -                docker load -i stage-artifacts/${repo}.tar
    -            else
    -                continue
    -            fi
    -        fi
```

if that makes sense

---

Yes, makes sense